core/database/api/repo_router.js

```diff
@@ -92,48 +143,26 @@ router.post('/create', function (req, res) {
                 if ( !client.is_admin )
                     throw g_lib.ERR_PERM_DENIED;
 
-                var obj = {
-                    capacity: req.body.capacity,
-                    pub_key: req.body.pub_key,
-                    address: req.body.address,
-                    endpoint: req.body.endpoint,
-                    path: req.body.path
-                };
+                var obj = parseRepoDoc( req.body );
 
                 g_lib.procInputParam( req.body, "id", false, obj );
-                g_lib.procInputParam( req.body, "title", false, obj );
-                g_lib.procInputParam( req.body, "summary", false, obj );
-                g_lib.procInputParam( req.body, "domain", false, obj );
-
-                if ( !obj.path.startsWith("/"))
-                    throw [ g_lib.ERR_INVALID_PARAM, "Repository path must be an absolute path file system path." ];
-
-                if ( !obj.path.endsWith("/"))
-                    obj.path += "/";
-
-                var idx = obj.path.lastIndexOf( "/", obj.path.length - 2 );
-                if ( obj.path.substr( idx + 1, obj.path.length - idx - 2 ) != obj._key )
-                    throw [ g_lib.ERR_INVALID_PARAM, "Last part of repository path must be repository ID suffix (" + obj._key + ")" ];
-
-                if ( req.body.exp_path ){
-                    obj.exp_path = req.body.exp_path;
-                    if ( !obj.exp_path.endsWith("/"))
-                        obj.path += "/";
-                }
 
                 var repo = g_db.repo.save( obj, { returnNew: true });
 
                 for ( var i in req.body.admins ) {
-                    if ( !g_db._exists( req.body.admins[i] ))
+                    if ( !g_db._exists( req.body.admins[i] )){
                         throw [g_lib.ERR_NOT_FOUND,"User, "+req.body.admins[i]+", not found"];
+                    }
```

Comment:
This is an improvement, I have hit several errors when I have accidentally deleted white space in front of code that had no curly braces.

Reply:
Yeah, I'm trying to fix these when I find them / think about it.